### PR TITLE
Table footnotes

### PR DESCRIPTION
**Pull request by oliver----**
## Why

Implementation of Feature 'Table Footnotes'. 

Related:
- #556

## What

- [x] Extend kitchen-sink **(O)**
- [x] Extend TextureArticleSchema to allow for `<table-wrap-foot>`
- [x] Extend `TableFigure` to allow for 0-n footnotes (-> children)
- [x] Implement Importer and create footnotes and populate `tableFig.footnotes`
- [x] Implement a label generator for table footnotes **(O)**
- [x] Refactor internal xref collection API for XrefEditDialog **(O)**  
- [x] Display footnotes and footnote references in TableFigure
- [x] Implement dialog for table-footnotes so that it allows only to choose from table footnotes
- [x] InsertFootnoteCrossReferenceCommand for both regular and table footnotes according to detected scope
- [x] Determine xpath for table selections (CustomSelection) 
- [x] Implement an InsertFootnoteCommand that creates a footnote in the according scope
- [x] Add InsertFootnoteTool to both views (currently only metadata)
- [x] After creating the first footnote xref, table- footnotes do not update anymore **(O)**
- [x] Fix TableFootnoteManager issues (e.g. no label for xrefs in caption) **(O)**
- [x] Add tests for table-footnote label generator **(O)**
- [x] Order footnotes in order of cross-refs (similar to how references are ordered) **(O)**
- [x] Implement exporter that stores created footnotes into `<table-wrap-foot>` **(O)**
- [x] Add a command and tool to remove a footnote (enabled when cursor inside fn)
- [x] Add tests for table-footnote interaction (create, delete, x-reference)
- [x] Fix table editor in Metadata mode
- [x] Add labels for table figures ('Label:', 'Title:', 'Caption'; 'Footnotes:') and regular figures styled in a same way as we have it in manuscript

### CODE REVIEW

**Comment by oliver----** [comment]
Note: please introduce a TableFigureModel (extends FigureModel) and a TableFigureComponent (extends FigureComponent)

**Comment by oliver----** [comment]
@Integral I have started a first iteration on the label generator (WIP)

**Comment by oliver----** [comment]
If you need the label for rendering, there is `tableFigure._tableFootnoteManager` which has `getLabel(pos)`

**Comment by oliver----** [comment]
For the EditXrefDialog I will have to do some refactorings. The current approach is not sufficient.

**Comment by Integral** [comment]
@oliver---- do we want to use collections for managing table footnotes? E.g. one collection for each figure? I see that with footnote manager we are already going this way, and I think that it we should do that the same with collections...

**Comment by Integral** [comment]
Nice. Also it would be easier for us to manage it.

**Comment by oliver----** [comment]
UI needs improvement: #1069

**Comment by oliver----** [comment]
Quick-fixed the Xref target resolver. At some point this needs a refactor.

**Comment by Integral** [comment]
screen shot 2018-12-05 at 13 37 09
We should do an iteration on metadata table. There are no fn renderers and wrong css classes now. Also tools should be available there. My suggestion would be to do it separately closer to test phase.

**Comment by oliver----** [comment]
@michael we need your action regarding a potential requirement for re-ordering of (table) footnotes.

File: src/article/editor/RemoveFootnoteCommand.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import RemoveItemCommand from './RemoveItemCommand'
+
+export default class RemoveCollectionItemCommand extends RemoveItemCommand {
```
Comment (oliver----) [inline src/article/editor/RemoveFootnoteCommand.js]: The file is called RemoveFootnoteCommand. Would like to rename the file or rename the class?

Comment (oliver----) [reply src/article/editor/RemoveFootnoteCommand.js]: I'd say, go for RemoveFootnoteCommand.
When we introduce a similar one we gonna extract a shared impl.

**Comment by oliver----** [comment]
@Integral very nice! Thanks for the tests!

**Comment by oliver----** [comment]
Perfect! 👏

**Comment by oliver----** [comment]
Let's get this merged! 🍾 🎉 🎆

**Comment by Integral** [comment]
Wonderful!

**Pull request merged**